### PR TITLE
Support SO_RXQ_OVFL socket option (android/fuchsia/linux)

### PR DESCRIPTION
**Pull request by junhochoi**
This PR implements support of RXQ_OVFL flag and parsing ControlMessage
to get the packet drop counter of UDP socket.

### CODE REVIEW

File: src/sys/socket/mod.rs [hunk 1]
```diff
@@ -826,6 +842,14 @@ pub enum ControlMessage<'a> {
               target_os = "android",
               target_os = "ios",))]
     Ipv6PacketInfo(&'a libc::in6_pktinfo),
+
+    // SO_RXQ_OVFL indicates that an unsigned 32 bit value
```
Comment (asomers) [inline src/sys/socket/mod.rs:846]: Did you mean to make this a doc comment instead of a regular comment?

File: src/sys/socket/mod.rs [hunk 2]
```diff
@@ -988,6 +1020,8 @@ impl<'a> ControlMessage<'a> {
               target_os = "netbsd", target_os = "freebsd",
               target_os = "android", target_os = "ios",))]
             ControlMessage::Ipv6PacketInfo(_) => libc::IPPROTO_IPV6,
+            #[cfg(target_os = "linux")]
+            ControlMessage::RxqOvfl(_) => libc::SO_RXQ_OVFL,
```
Comment (asomers) [inline src/sys/socket/mod.rs:1024]: Could you double-check this part?

**Review by asomers — CHANGES_REQUESTED** [review]
Don't forget to add a CHANGELOG entry.  Also, what about Android and Fuchsia?  libc defines this constant for them.

File: src/sys/socket/mod.rs (hunk 1 above)
Comment (junhochoi) [reply src/sys/socket/mod.rs]: Changed to `///`.

File: src/sys/socket/mod.rs (hunk 2 above)
Comment (junhochoi) [reply src/sys/socket/mod.rs:1024]: I updated the condition to include `android`, `fuchsia`, `linux`. CI seems running ok.

**Comment by junhochoi** [comment]
> Don't forget to add a CHANGELOG entry. Also, what about Android and Fuchsia? libc defines this constant for them.

I added CHANGELOG entry and addressed other comments. Thanks!

File: src/sys/socket/mod.rs (hunk 2 above)
Comment (asomers) [reply src/sys/socket/mod.rs:1024]: I mean that `SO_RXQ_OVFL` doesn't look like a valid value for `cmsg_level`.  Is that really correct?

Comment (junhochoi) [reply src/sys/socket/mod.rs]: Oh right. updated to `libc::SOL_SOCKET`. thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,9 @@ This project adheres to [Semantic Versioning](https://semver.org/).
   (#[1456](https://github.com/nix-rust/nix/pull/1456))
 - Added `TcpUserTimeout` socket option (sockopt) on Linux and Fuchsia.
   (#[1457](https://github.com/nix-rust/nix/pull/1457))
+  (#[1447](https://github.com/nix-rust/nix/pull/1447))
```
Comment (asomers) [inline CHANGELOG.md:15]: Somehow you duped a line here.
```suggestion
```

**Review by asomers — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (junhochoi) [reply CHANGELOG.md]: Fixed (I don't know where it comes from! probably something during rebase with master) Thanks

**Review by asomers — APPROVED** [review]
bors r+

**Comment by asomers** [comment]
The tests are failing due to a just-landed PR on head.

**Comment by junhochoi** [comment]
> The tests are failing due to a just-landed PR on head.

rebased to master and pushed a fix.

**Comment by junhochoi** [comment]
@asomers the build is green. should be good to go.

**Review by asomers — APPROVED** [review]
bors r+

**Comment by asomers** [comment]
bors retry

**Comment by junhochoi** [comment]
rebased to master again.

**Review by asomers — APPROVED** [review]
bors r+

**Pull request merged**